sys/arduino/include/spiport.hpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_arduino_api
+ * @{
+ *
+ * @file
+ * @brief       Definition of the Arduino 'SPI' interface
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#ifndef SPIPORT_H
+#define SPIPORT_H
+
+extern "C" {
+#include "arduino_board.h"
+#include "byteorder.h"
+#include "periph/spi.h"
+}
+
+/**
+ * @name    Arduino compatible SPI modes
+ * @{
+ */
+#define SPI_MODE0       (0)     /**< CPOL=0, CPHA=0 */
+#define SPI_MODE1       (1)     /**< CPOL=0, CPHA=1 */
+#define SPI_MODE2       (2)     /**< CPOL=1, CPHA=0 */
+#define SPI_MODE3       (3)     /**< CPOL=1, CPHA=1 */
+/** @} */
+
+/**
+ * @name    Arduino compatible bit order values for SPI
+ * @{
+ */
+#define MSBFIRST        (0)     /**< most significat bit first */
+/** @} */
+
+/**
+ * @brief   Arduino SPI configuration interface
+ */
+class SPISettings {
+    public:
+    /**
```

Comment:
One question. What is the encoding style for indenting `private`, `public`, ... keywords and their members? I have never seen them have the same indentation.

Reply:
Don't ask me, I'm a C developer ;-) If you prefer a specific style, I'm happy to update the code to match it.

Reply:
I was programming C++ for more than 15 years. We used indention of 2 characters and a style like
```
class ...
{
  public:
    member

  private:
    member
};
```
What I have often seen, e.g., in g++ includes is
```
class
{
public:
  member

private:
  member
};
```
If RIOT doesn't define a style for C++, we should probably use the style like in standard C++ includes with an indention of 4 characters.

Reply:
Have you thought about the style.